core/dbt/contracts/graph/nodes.py

```diff
@@ -552,6 +540,30 @@ def depends_on_macros(self):
         return self.depends_on.macros
 
 
+@dataclass
+class FileSlice(dbtClassMixin, Replaceable):
+    """Provides file slice level context about what something was created from.
+
+    Implementation of the dbt-semantic-interfaces `FileSlice` protocol
+    """
+
+    filename: str
+    content: str
+    start_line_number: int
+    end_line_number: int
+
+
+@dataclass
+class Metadata(dbtClassMixin, Replaceable):
```

Comment:
We already have a Metadata class in model_config.py.  Could this be given a more descriptive name?

Reply:
@QMalcolm What do you think about calling this class SemanticMetadata? I don't think it should affect serialization.

Reply:
I think that'd be a great distinction to make. And yea the class name shouldn't affect the serialization 👍

Reply:
Done. I chose the name SourceFileMetadata.